### PR TITLE
Secure Boot Suite

### PR DESCRIPTION
**Pull request by filipleple**

### CODE REVIEW

**Comment by filipleple** [comment]
pre-commit.ci autofix

**Comment by filipleple** [comment]
pre-commit.ci autofix

**Comment by filipleple** [comment]
pre-commit.ci autofix

**Comment by filipleple** [comment]
pre-commit.ci autofix

File: lib/secure-boot-lib.robot [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    ...    been reset since flashing.
+    Enter Setup Menu Tianocore
+    Enter Device Manager Submenu
+    Enter Secure Boot Configuration Submenu
+    Press Key N Times And Enter    3    ${ARROW_DOWN}
+    Read From Terminal Until    INFO
+    Press Key N Times    1    ${ENTER}
+    Save Changes And Reset    2    5
+
+Upload Required Images
```
Comment (macpijan) [inline lib/secure-boot-lib.robot:44]: This keyword is not used

File: lib/secure-boot-lib.robot [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    SBO001.001 not supported
+    Skip If    not ${TESTS_IN_UBUNTU_SUPPORT}    SBO001.001 not supported
+    Power On
+    Enter Setup Menu Tianocore
+    Enter Device Manager Submenu
+    Enter Secure Boot Configuration Submenu
+    ${attempt_sb_can_be_selected}=    Check If Attempt Secure Boot Can Be Selected
+    IF    not ${attempt_sb_can_be_selected}
+        Reenter Menu
+        Press Key N Times And Enter    2    ${ARROW_DOWN}
```
Comment (macpijan) [inline lib/secure-boot-lib.robot:81]: @filipleple We should not be hardcoding menu positions.

File: dasharo-security/secure-boot.robot [hunk 3]
```diff
@@ -13,11 +13,16 @@ Resource            ../rtectrl-rest-api/rtectrl.robot
...
-# - go threough them and make sure they are doing what the name suggest (not
-# exactly the case right now)
+Resource            ../lib/secure-boot-lib.robot
+
+# Required setup keywords:
+# Prepare Test Suite - elementary setup keyword for all tests.
+# Reset Secure Boot Keys For The First Time - resets secure boot keys
+#    for the first time since flashing. Otherwise enabling secure boot
+#    is not possible.
+# Upload Required Images - uploads all required files onto the PiKVM.
```
Comment (macpijan) [inline dasharo-security/secure-boot.robot:23]: @filipleple  Why is this commented out? How we make sure that required images are on the PiKVM? What happens if we enable it and try to upload every time we run the suite?

File: lib/secure-boot-lib.robot [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    ...    It can also take the fileformat parameter
+    ...    indicating whether the file is of a valid format.
+    ...    If not, it will look for the appropriate ERROR
+    ...    message.
+    [Arguments]    ${cert_filename}    ${fileformat}=GOOD
+    Enter Setup Menu Tianocore
+    Enter Device Manager Submenu
+    Enter Secure Boot Configuration Submenu
+    Enter Custom Secure Boot Options
+    Enroll Certificate    ${cert_filename}    ${fileformat}
```
Comment (macpijan) [inline lib/secure-boot-lib.robot:14]: @filipleple this keyword is not defined

File: dasharo-security/secure-boot.robot (hunk 3 above)
Comment (filipleple) [reply dasharo-security/secure-boot.robot:23]: @macpijan Right, I put it in the Suite Setup/Run Reywords command. 
Thankfully, we can just attempt uploading them every time - the PiKVM handles checking whether they're already there or not on its own.

File: lib/secure-boot-lib.robot (hunk 4 above)
Comment (filipleple) [reply lib/secure-boot-lib.robot:14]: Resolved by recent commit

File: lib/secure-boot-lib.robot (hunk 1 above)
Comment (filipleple) [reply lib/secure-boot-lib.robot:137]: Now used in Suite Setup

File: lib/secure-boot-lib.robot (hunk 2 above)
Comment (filipleple) [reply lib/secure-boot-lib.robot]: Resolved in recent commit

Comment (macpijan) [reply lib/secure-boot-lib.robot]: @filipleple It would be better to provide commit ID when saying this

File: dasharo-security/secure-boot.robot [hunk 5]
```diff
@@ -9,19 +9,22 @@ Library             RequestsLibrary
 # TODO: maybe have a single file to include if we need to include the same
 # stuff in all test cases
 Resource            ../sonoff-rest-api/sonoff-api.robot
+Resource            ../pikvm-rest-api/pikvm_comm.robot
 Resource            ../rtectrl-rest-api/rtectrl.robot
 Resource            ../variables.robot
 Resource            ../keywords.robot
 Resource            ../keys.robot
+Resource            ../lib/secure-boot-lib.robot
```
Comment (macpijan) [inline dasharo-security/secure-boot.robot:17]: This can go into keywords.robot

File: dasharo-security/secure-boot.robot [hunk 6]
```diff
@@ -9,19 +9,22 @@ Library             RequestsLibrary
 # TODO: maybe have a single file to include if we need to include the same
 # stuff in all test cases
 Resource            ../sonoff-rest-api/sonoff-api.robot
+Resource            ../pikvm-rest-api/pikvm_comm.robot
```
Comment (macpijan) [inline dasharo-security/secure-boot.robot:12]: This as well (maybe already there?)

File: dasharo-security/secure-boot.robot [hunk 7]
```diff
@@ -89,72 +92,67 @@ SBO002.002 UEFI Secure Boot (Windows 11)
...
+    Enter UEFI Shell And Boot .EFI File    hello-bad-keys.efi    Access Denied
+
+SBO006.001 Reset Secure Boot Keys option availability (firmware)
+    [Documentation]    This test verifies that the Reset Secure Boot Keys
+    ...    option is available
+    Skip If    not ${SECURE_BOOT_SUPPORT}    SBO006.001 not supported
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    SBO006.001 not supported
+    Skip If    not ${TESTS_IN_UBUNTU_SUPPORT}    SBO006.001 not supported
+    Power On
+    Verify Reset Secure Boot Keys
```
Comment (macpijan) [inline dasharo-security/secure-boot.robot:134]: I would remove this KW and use directly as we do in other tests

```
    Enter Setup Menu Tianocore
    Enter Device Manager Submenu
    Enter Secure Boot Configuration Submenu
    Read From Terminal Until    Reset Secure Boot Keys
```

File: lib/secure-boot-lib.robot [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+
+Verify Reset Secure Boot Keys
+    [Documentation]    Verifies that Reset Secure Boot Keys
+    ...    option is available.
+    Enter Setup Menu Tianocore
+    Enter Device Manager Submenu
+    Enter Secure Boot Configuration Submenu
+    Read From Terminal Until    Reset Secure Boot Keys
+
+Reset Secure Boot Keys Again
```
Comment (macpijan) [inline lib/secure-boot-lib.robot:141]: Why do we need this strange `Reset Secure Boot Keys **Again**` KW? We simply need KW to reset keys, that's it.
Like I asked to just copy from: https://github.com/Dasharo/open-source-firmware-validation/commit/7e69d7ec371ec73d1276830bc0108cb862b71cb4#diff-b06ff90d8f021dfa2e18d217f3d4c1e6861d6f671893cf292b6718b80227921aR71

Comment (filipleple) [reply lib/secure-boot-lib.robot]: Resolved in [21a0c1f](https://github.com/Dasharo/open-source-firmware-validation/pull/17/commits/21a0c1f4bc636ec5ee87d9cd923764a084c70656)

File: dasharo-security/secure-boot.robot (hunk 7 above)
Comment (filipleple) [reply dasharo-security/secure-boot.robot]: Resolved in [21a0c1f](https://github.com/Dasharo/open-source-firmware-validation/pull/17/commits/21a0c1f4bc636ec5ee87d9cd923764a084c70656)

File: dasharo-security/secure-boot.robot (hunk 6 above)
Comment (filipleple) [reply dasharo-security/secure-boot.robot]: Resolved in [21a0c1f](https://github.com/Dasharo/open-source-firmware-validation/pull/17/commits/21a0c1f4bc636ec5ee87d9cd923764a084c70656)

File: dasharo-security/secure-boot.robot (hunk 5 above)
Comment (filipleple) [reply dasharo-security/secure-boot.robot]: Resolved in [21a0c1f](https://github.com/Dasharo/open-source-firmware-validation/pull/17/commits/21a0c1f4bc636ec5ee87d9cd923764a084c70656)

**Pull request merged**